mods/cnc/maps/nod05/nod05.lua

```diff
@@ -0,0 +1,319 @@
+
+GDIStartUnits = {}
+
+XxxxTriggerFunction = function()
+	if not Radar.IsDead then
+		local list = Nod.GetGroundAttackers()
+		AirstrikeProxy = Actor.Create("airstrike.proxy", true, { Owner = GDI })
+		AirstrikeProxy.SendAirstrike(list[DateTime.GameTime % #list + 1].CenterPosition, false, 256 - 28)
+		AirstrikeProxy.Destroy()
+		Trigger.AfterDelay(XxxxTriggerFunctionTime, XxxxTriggerFunction)
+	end
+end
+
+YyyyTriggerFunction = function()
+	if not YyyyTriggerSwitch then
+		for type, count in pairs(Gdi2Units) do
+			MyActors = Utils.Take(count, GDI.GetActorsByType(type))
+			Utils.Do(MyActors, function(actor)
+				WaypointMovementAndHunt(actor, Gdi2Waypoints)
+			end)
+		end
+	end
+end
+
+ZzzzTriggerFunction = function()
+	if not ZzzzTriggerSwitch then
+		for type, count in pairs(Gdi1Units) do
+			MyActors = Utils.Take(count, GDI.GetActorsByType(type))
+			Utils.Do(MyActors, function(actor)
+				WaypointMovementAndHunt(actor, Gdi1Waypoints)
+			end)
+		end
+	end
+end
+
+Grd1TriggerFunction = function()
+	MyActors = Utils.Take(2, GDI.GetActorsByType('jeep'))
+	Utils.Do(MyActors, function(actor)
+		WaypointMovementAndHunt(actor, Gdi5Waypoints)
+	end)
+end
+
+Atk5TriggerFunction = function()
+	WaypointMovementAndHunt(GDI.GetActorsByType('mtnk')[1], Gdi12Waypoints)
+end
+
+Atk2TriggerFunction = function()
+	for type, count in pairs(Gdi1Units) do
+		MyActors = Utils.Take(count, GDI.GetActorsByType(type))
+		Utils.Do(MyActors, function(actor)
+			WaypointMovementAndHunt(actor, Gdi1Waypoints)
+		end)
+	end
+end
+
+Atk3TriggerFunction = function()
+	for type, count in pairs(Gdi2Units) do
+		MyActors = Utils.Take(count, GDI.GetActorsByType(type))
+		Utils.Do(MyActors, function(actor)
+			WaypointMovementAndHunt(actor, Gdi2Waypoints)
+		end)
+	end
+end
+
+Atk4TriggerFunction = function()
+	WaypointMovementAndHunt(GDI.GetActorsByType('jeep')[1], Gdi3Waypoints)
+end
+
+Atk6TriggerFunction = function()
+	WaypointMovementAndHunt(GDI.GetActorsByType('mtnk')[1], Gdi2Waypoints)
+end
+
+Atk1TriggerFunction = function()
+	Reinforcements.ReinforceWithTransport(GDI, 'tran', GDIReinforceUnits, { waypoint9.Location, waypoint26.Location }, nil,
+	function(transport, cargo)
+		transport.UnloadPassengers()
+		Utils.Do(cargo, function(actor)
+			IdleHunt(actor)
+		end)
+	end)
+end
+
+AutoTriggerFunction = function()
+	local units = AllUnits[DateTime.GameTime % #AllUnits + 1]
+	local waypoints = AllWaypoints[DateTime.GameTime % #AllWaypoints + 1]
+
+	for type, count in pairs(units) do
+		MyActors = Utils.Take(count, GDI.GetActorsByType(type))
+		Utils.Do(MyActors, function(actor)
+			WaypointMovementAndHunt(actor, waypoints)
+		end)
+	end
+end
+
+HuntTriggerFunction = function()
+	local list = GDI.GetGroundAttackers()
+	Utils.Do(list, function(unit)
+		IdleHunt(unit)
+	end)
+end
+
+WaypointMovementAndHunt = function(unit, waypoints)
+	Utils.Do(waypoints, function(waypoint)
+		unit.AttackMove(waypoint.Location)
```

Comment:
We need a nil check somewhere around here:
`Eluant.LuaException: [string "nod05.lua"]:144: attempt to index upvalue 'unit' (a nil value)`
(Probably `GetActorsByType` returned nil as there were no actors to get.)